opensrp/fhircore
# Add latest 0.2.4 fixes to Main(Temporary )

author Elly Kitoto <junkmailstoelly@gmail.com> 1688737333 +0300 committer Elly Kitoto <junkmailstoelly@gmail.com> 1688744548 +0300

parent e3bf91f440c4ca1f180a6ab2310479f1ae026db5
author Elly Kitoto <junkmailstoelly@gmail.com> 1688737333 +0300 committer Elly Kitoto <junkmailstoelly@gmail.com> 1688744349 +0300

Add latest fixes

- Fix rules engine sort resources function
- Fix refreshing profile data
- Handle questionnaire on submission actions
- Fix Unit tests
- Fix updating Task status to READY
- Upgrade JEXL version to v3.2.1
- Fix sorting resources via rules engine
- Revert conditional check for dependent task when marking Task as due



* Resolve list view not refreshing data



* Handle questionnaire on submission actions



* Fix failing tests



* Update android/quest/src/main/java/org/smartregister/fhircore/quest/ui/profile/ProfileFragment.kt



* Update android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt



* Update android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt



* Update android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskUtil.kt



* Fix Build 💚
- Fix Failing tests
- Add unit tests ✅

* Fix Build 💚

* Fix updating Task status to READY



---------

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] or Closes #[issue number]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._


**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app
